Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs

```diff
@@ -324,125 +302,84 @@ private async Task<HttpResponseInfo> SendAsyncInternal(HttpRequestOptions option
 
             options.CancellationToken.ThrowIfCancellationRequested();
 
-            var client = GetHttpClient(options.Url, options.EnableHttpCompression);
+            var client = GetHttpClient(options.Url);
 
             var httpWebRequest = GetRequestMessage(options, httpMethod);
 
-            if (options.RequestContentBytes != null ||
-                !string.IsNullOrEmpty(options.RequestContent) ||
-                httpMethod == HttpMethod.Post)
+            if (options.RequestContentBytes != null
+                || !string.IsNullOrEmpty(options.RequestContent)
+                || httpMethod == HttpMethod.Post)
             {
-                try
-                {
-                    httpWebRequest.Content = new StringContent(Encoding.UTF8.GetString(options.RequestContentBytes) ?? options.RequestContent ?? string.Empty);
 
-                    var contentType = options.RequestContentType ?? "application/x-www-form-urlencoded";
-
-                    if (options.AppendCharsetToMimeType)
-                    {
-                        contentType = contentType.TrimEnd(';') + "; charset=\"utf-8\"";
-                    }
-
-                    httpWebRequest.Headers.Add(HeaderNames.ContentType, contentType);
-                    await client.SendAsync(httpWebRequest).ConfigureAwait(false);
+                if (options.RequestContentBytes != null)
+                {
+                    httpWebRequest.Content = new ByteArrayContent(options.RequestContentBytes);
+                }
+                else if (options.RequestContent != null)
+                {
+                    httpWebRequest.Content = new StringContent(options.RequestContent);
+                }
+                else
+                {
+                    httpWebRequest.Content = new ByteArrayContent(Array.Empty<byte>());
                 }
-                catch (Exception ex)
+                /*
+                var contentType = options.RequestContentType ?? "application/x-www-form-urlencoded";
```

Comment:
Any reason not to remove this?

Reply:
Content types aren't being set atm (they are, but not as requested), this is there as a reminder (for now).

Reply:
Maybe add a TODO to make it searchable?